### DISCUSSION TITLE
chat_history always empty when using API

### DISCUSSION THREAD

**Discussion by ori-aventure** [FlowiseAI/Flowise#2034] [Q&A] [answered]
Hi, my chatflows work just fine when using the Flowise backend. But when I use the same flow through the API, the **chat_history** variable is always **empty**. What could this be? Am I doing session management wrong? What part of the API config am I missing? Thanks for reading, any help is much appreciated!
![debug_screenshot](https://github.com/FlowiseAI/Flowise/assets/152312095/58aa897d-42b2-43c9-8748-7ad4097c29f2)

**Comment by ori-aventure** [comment, marked as answer]
I guess I have to trace the messages in an array and provide the history in every API call payload myself... This is more complicated than expected. Is there an easier fix for this? I feel Flowise should be able to handle this for me. Thanks!!

https://docs.flowiseai.com/integrations/langchain/memory/short-term-memory

**Reply by HenryHengZJ** [reply]
We are working on it!

**Comment by sirsimonson** [comment]
Could we then reuse chat_history through flows when we set the correct sessionId and chatId on the API call?
